tests/potential/test_potential.py

```diff
@@ -308,5 +309,272 @@ def test_fromfile(self):
 
         temp.close()
 
+class test_PairPotential(unittest.TestCase):
+    """Unit tests for potential.PairPotential"""
+
+    class LinPot(potential.PairPotential):
+        """Linear potential function used to test potential.PairPotential"""
+        def __init__(self, types, params, default={}):
+            super().__init__(types, params, default)
+
+        def _energy(self, r, m, **params):
+            r,u,s = self._zeros(r)
+            u = m*r
+            if s:
+                u = u.item()
+            return u
+
+        def _force(self, r, m, **params):
+            r,f,s = self._zeros(r)
+            f = -m
+            if s:
+                f = f.item()
+            return f
+
+        def _derivative(self, param, r, **params):
+            r,d,s = self._zeros(r)
+            if param == 'm':
+                d = r
+            if s:
+                d = d.item()
+            return d
```

Comment:
You're probably going to need this class to test the `Tabulator` later. You can refer to it there as `test_PairPotential.LinPot`, or you can move this class outside the unittest, so it is accessible directly as `LinPot`.